source/s3_list_objects.c

```diff
@@ -142,7 +142,7 @@ static bool s_on_list_bucket_result_node_encountered(
         }
 
         if (ret_val && operation_data->on_object) {
-            ret_val |= operation_data->on_object(&fs_wrapper.fs_info, operation_data->user_data);
+            ret_val &= operation_data->on_object(&fs_wrapper.fs_info, operation_data->user_data);
```

Comment:
trivial: rename ~ret_val~ -> `keep_going` or something, so we don't confuse it with an `int` again

Reply:
or just do `keep_going = ` instead of this bit nonsense